recipes/symengine/all/conanfile.py

```diff
@@ -0,0 +1,91 @@
+from conans import ConanFile, CMake, tools
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class SymengineConan(ConanFile):
+    name = "symengine"
+    description = "A fast symbolic manipulation library, written in C++"
+    license = "MIT"
+    topics = ("symbolic", "algebra")
+    homepage = "https://symengine.org/"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "integer_class": ["boostmp", "gmp"],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "integer_class": "boostmp",
+    }
+    short_paths = True
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def requirements(self):
+        if self.options.integer_class == "boostmp":
+            self.requires("boost/1.76.0")
+        else:
+            self.requires("gmp/6.2.1")
+
+    def source(self):
+        tools.get(
+            **self.conan_data["sources"][self.version],
+            strip_root=True,
+            destination=self._source_subfolder,
+        )
+
+    def _configure_cmake(self):
+        if self._cmake is None:
+            self._cmake = CMake(self)
+            self._cmake.definitions["BUILD_TESTS"] = False
+            self._cmake.definitions["BUILD_BENCHMARKS"] = False
+            self._cmake.definitions["INTEGER_CLASS"] = self.options.integer_class
+            self._cmake.definitions["MSVC_USE_MT"] = False
```

Comment:
Should this be allowed by conan settings?

```suggestion
            self._cmake.definitions["MSVC_USE_MT"] = "MT" in self.settings.compiler.get_safe("runtime", "")
```

Reply:
It is fine.
What this option does is replace all occurrences of `MD` with `MT` in `CMAKE_C_FLAGS`.
But since the cmake helper already set it correctly, it is a no-op.
You suggestion is off course fine too.